Source/Scene/AutoExposure.js

```diff
@@ -0,0 +1,416 @@
+                    height : height,
+                    pixelFormat : pixelFormat,
+                    pixelDatatype : pixelDatatype,
+                    sampler : sampler
+                })]
+            });
+        }
+
+        var lastTexture = framebuffers[length - 1].getColorTexture(0);
+        autoexposure._previousLuminance = new Framebuffer({
+            context : context,
+            colorTextures : [new Texture({
+                context : context,
+                width : lastTexture.width,
+                height : lastTexture.height,
+                pixelFormat : pixelFormat,
+                pixelDatatype : pixelDatatype,
+                sampler : sampler
+            })]
+        });
+
+        autoexposure._framebuffers = framebuffers;
+    }
+
+    function destroyCommands(autoexposure) {
+        var commands = autoexposure._commands;
+        if (!defined(commands)) {
+            return;
+        }
+
+        var length = commands.length;
+        for (var i = 0; i < length; ++i) {
+            commands[i].shaderProgram.destroy();
+        }
+        autoexposure._commands = undefined;
+    }
+
+    function createUniformMap(autoexposure, index) {
+        var uniforms;
+        if (index === 0) {
+            uniforms = {
+                colorTexture : function() {
+                    return autoexposure._colorTexture;
+                },
+                colorTextureDimensions : function() {
+                    return autoexposure._colorTexture.dimensions;
+                }
+            };
+        } else {
+            var texture = autoexposure._framebuffers[index - 1].getColorTexture(0);
+            uniforms = {
+                colorTexture : function() {
+                    return texture;
+                },
+                colorTextureDimensions : function() {
+                    return texture.dimensions;
+                }
+            };
+        }
+
+        uniforms.minimumLuminance = function() {
+            return autoexposure.minimumLuminance;
+        };
+        uniforms.maximumLuminance = function() {
+            return autoexposure.maximumLuminance;
+        };
+        uniforms.previousLuminance = function() {
+            return autoexposure._previousLuminance.getColorTexture(0);
+        };
+
+        return uniforms;
+    }
+
+    function getShaderSource(index, length) {
+        var source =
+            'uniform sampler2D colorTexture; \n' +
+            'varying vec2 v_textureCoordinates; \n' +
+            'float sampleTexture(vec2 offset) { \n';
+
+        if (index === 0) {
+            source +=
+                '    vec4 color = texture2D(colorTexture, v_textureCoordinates + offset); \n' +
+                '    return czm_luminance(color.rgb); \n';
+        } else {
+            source +=
+                '    return texture2D(colorTexture, v_textureCoordinates + offset).r; \n';
+        }
+
+        source += '}\n\n';
+
+        source +=
```

Comment:
The answer may be no, but it is possible and reasonable to pull the parallel reduction core into a standalone `czm_` function that we may use elsewhere, e.g., for edge-preserving blur?